src/Features/OnboardingTasks/TaskList.php

```diff
@@ -112,12 +112,27 @@ public function add_task( $args ) {
 		$this->tasks[] = new Task( $args );
 	}
 
+	/**
+	 * Get only visible tasks in list.
+	 */
+	public function get_viewable_tasks() {
+		return array_values(
+			array_filter(
+				$this->tasks,
+				function( $task ) {
+					return $task->can_view;
+				}
+			)
+		);
+	}
+
 	/**
 	 * Get the list for use in JSON.
 	 *
+	 * @param boolean $only_viewable Only display visible tasks.
 	 * @return array
 	 */
-	public function get_json() {
+	public function get_json( $only_viewable = false ) {
```

Comment:
Passing arguments to filter tasks feels a bit like an anti-pattern.  

Maybe we could create a method to allow filtering of tasks and keep the `get_json()` method simple:

```
$task_list->filter_tasks( array( 'can_view' => true ) );
$task_list->get_json()
```

Reply:
Thinking about this more, one other approach if we want to keep things simple would be to create a `get_viewable_tasks()` method on `TaskList`.  I'd be okay if `get_json()` always used `get_viewable_tasks()` given it's use is for client-side JSON.

Reply:
I had actually already created a `get_viewable_tasks()` method (just above this comment), but I removed the parameter for `get_json` in a53f441620ce1dc4c788970411fe9512255c9ade. 

The only downside here is that it removes the ability to show non-viewable tasks from the endpoint at all, but may not be a need going forward so we can keep it simple.